### PR TITLE
[Refactor] Rename DeviceTestTask to TestRun

### PR DESCRIPTION
**Pull request by hydraxman**
rename的主要目的是把概念定义好 一个testRun不再绑定某一台设备 所以把名字里的device去掉，为真正的cross-platform做准备。
Expect no changes on the DB schema and JSON protocols. And all changes happen within the scope of the code.

### CODE REVIEW

File: common/src/main/java/com/microsoft/hydralab/common/entity/common/TestRun.java [hunk 1]
```diff
@@ -18,9 +17,9 @@
 
 @Data
 @Entity
-@Table(indexes = {
-        @Index(name = "task_id_index", columnList = "test_task_id", unique = false)})
-public class DeviceTestTask implements Serializable {
+@Table(name = "device_test_task", indexes = {
```
Comment (hydraxman) [inline common/src/main/java/com/microsoft/hydralab/common/entity/common/TestRun.java:20]: @zhou9584 please help check whether the table name is correct, thanks!

Comment (zhou9584) [reply common/src/main/java/com/microsoft/hydralab/common/entity/common/TestRun.java:20]: The table name is correct. We can also transfer the data from old table to the new table when deploying new package.

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java [hunk 2]
```diff
@@ -96,68 +96,68 @@ protected DeviceTestTask buildDeviceTestTask(DeviceInfo deviceInfo, TestTask tes
             }
         }
 
-        deviceTestTask.setDeviceTestResultFolder(deviceTestResultFolder);
-        Logger loggerForDeviceTestTask = createLoggerForDeviceTestTask(deviceTestTask, testTask.getTestSuite(), parentLogger);
-        deviceTestTask.setLogger(loggerForDeviceTestTask);
-        testTask.addTestedDeviceResult(deviceTestTask);
-        return deviceTestTask;
+        testRun.setTestRunResultFolder(deviceTestResultFolder);
+        Logger loggerForDeviceTestTask = createLoggerForDeviceTestTask(testRun, testTask.getTestSuite(), parentLogger);
```
[the remark this answers is not in the document]
Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java:100]: Good catches, will update.

File: common/src/main/java/com/microsoft/hydralab/common/entity/common/TestRun.java (hunk 1 above)
Comment (hydraxman) [reply common/src/main/java/com/microsoft/hydralab/common/entity/common/TestRun.java:20]: Thanks, please help resolve this conversation if you think it's fine.

Comment (hydraxman) [reply common/src/main/java/com/microsoft/hydralab/common/entity/common/TestRun.java:20]: @zhou9584

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (hydraxman) [reply agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java]: Done, resolved.

File: common/src/main/java/com/microsoft/hydralab/common/repository/DeviceTestResultRepository.java [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
+import com.microsoft.hydralab.common.entity.common.TestRun;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 import java.util.Optional;
 
 @Repository
-public interface DeviceTestResultRepository extends JpaRepository<DeviceTestTask, String> {
+public interface DeviceTestResultRepository extends JpaRepository<TestRun, String> {
```
Comment (zhou9584) [inline common/src/main/java/com/microsoft/hydralab/common/repository/DeviceTestResultRepository.java:13]: Shall we rename this file?

Comment (hydraxman) [reply common/src/main/java/com/microsoft/hydralab/common/repository/DeviceTestResultRepository.java]: Updated.

**Review by zhou9584 — DISMISSED** [review]

**Review by taoran6 — DISMISSED** [review]

File: common/src/main/java/com/microsoft/hydralab/common/entity/common/AndroidTestUnit.java [hunk 4]
```diff
@@ -55,7 +55,7 @@ public class AndroidTestUnit implements Serializable {
     private transient File cpuTraceFile;
 
     @Transient
-    private DeviceTestTask deviceTestTask;
+    private TestRun testRun;
```
Comment (hydraxman) [inline common/src/main/java/com/microsoft/hydralab/common/entity/common/AndroidTestUnit.java]: This change will impact json, please revert.

**Review by taoran6 — APPROVED** [review]

**Pull request merged**